pkg/collector/corechecks/containers/cri/check_test.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build cri
+// +build cri
+
+package cri
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/generic"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/cri"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/cri/crimock"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+
+	criTypes "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCriCheck(t *testing.T) {
+	// Creating mocks
+	fullStatContainer := generic.CreateContainerMeta("containerd", "cID100")
+	fullStatContainer.Labels = map[string]string{"io.kubernetes.pod.namespace": "default"}
+	containersMeta := []*workloadmeta.Container{
+		// Container with full stats
+		fullStatContainer,
+		// Should never been called as it does not have the kubernetes label
+		generic.CreateContainerMeta("docker", "cID101"),
+	}
+
+	containersStats := map[string]metrics.MockContainerEntry{
+		"cID100": metrics.GetFullSampleContainerEntry(),
+		"cID101": metrics.GetFullSampleContainerEntry(),
+	}
+
+	// Inject mock processor in check
+	mockCri := &crimock.MockCRIClient{}
+	mockSender, processor, _ := generic.CreateTestProcessor(containersMeta, nil, containersStats, metricsAdapter{}, getProcessorFilter(nil))
+	processor.RegisterExtension("cri-custom-metrics", &criCustomMetricsExtension{criGetter: func() (cri.CRIClient, error) { return mockCri, nil }})
+
+	mockCri.On("ListContainerStats").Return(map[string]*criTypes.ContainerStats{
+		"cID100": {
+			WritableLayer: &criTypes.FilesystemUsage{
+				UsedBytes: &criTypes.UInt64Value{
+					Value: 10,
+				},
+				InodesUsed: &criTypes.UInt64Value{
+					Value: 20,
+				},
+			},
+		},
+		"cID101": {
+			WritableLayer: &criTypes.FilesystemUsage{
+				UsedBytes: &criTypes.UInt64Value{
+					Value: 30,
+				},
+				InodesUsed: &criTypes.UInt64Value{
+					Value: 40,
+				},
+			},
+		},
+	}, nil)
+
+	// Create CRI check
+	check := CRICheck{
+		instance: &CRIConfig{
+			CollectDisk: true,
+		},
+		processor: *processor,
+	}
+
+	err := check.runProcessor(mockSender)
+	assert.NoError(t, err)
+
+	expectedTags := []string{"runtime:containerd"}
+	mockSender.AssertNumberOfCalls(t, "Rate", 1)
+	mockSender.AssertNumberOfCalls(t, "Gauge", 4)
+
+	mockSender.AssertMetricInRange(t, "Gauge", "cri.uptime", 0, 600, "", expectedTags)
+	mockSender.AssertMetric(t, "Rate", "cri.cpu.usage", 100, "", expectedTags)
+	mockSender.AssertMetric(t, "Gauge", "cri.mem.rss", 300, "", expectedTags)
```

Comment:
It might be good for readability to assign `metrics.GetFullSampleContainerEntry()` to a variable so can refer to it instead of defining the constants 100 and 300 here. I'm pointing this out because right now these 2 constants are in a different file and it might be confusing to know where they're coming from.

Reply:
I tend to always put hardcoded values as `expected` in unit tests, to make sure any change is intended and understood.